service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/EnableVmLoggingStep.java

```diff
@@ -0,0 +1,187 @@
+import com.azure.resourcemanager.compute.ComputeManager;
+import com.azure.resourcemanager.compute.models.VirtualMachine;
+import com.azure.resourcemanager.monitor.MonitorManager;
+import com.azure.resourcemanager.monitor.fluent.models.DataCollectionRuleAssociationProxyOnlyResourceInner;
+import io.opencensus.contrib.spring.aop.Traced;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EnableVmLoggingStep implements Step {
+
+  private static final Logger logger = LoggerFactory.getLogger(EnableVmLoggingStep.class);
+  public static final String DATA_COLLECTION_RULES_TYPE = "Microsoft.Insights/dataCollectionRules";
+
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureVmResource resource;
+  private final LandingZoneApiDispatch landingZoneApiDispatch;
+  private final AuthenticatedUserRequest userRequest;
+
+  public EnableVmLoggingStep(
+      AzureConfiguration azureConfig,
+      CrlService crlService,
+      ControlledAzureVmResource resource,
+      LandingZoneApiDispatch landingZoneApiDispatch,
+      AuthenticatedUserRequest userRequest) {
+    this.azureConfig = azureConfig;
+    this.crlService = crlService;
+    this.resource = resource;
+    this.landingZoneApiDispatch = landingZoneApiDispatch;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    getDataCollectionRuleFromLandingZone()
+        .ifPresent(
+            dcr -> {
+              final String vmId =
+                  context.getWorkingMap().get(AzureVmHelper.WORKING_MAP_VM_ID, String.class);
+
+              addMonitorAgentToVm(context, vmId);
+
+              createDataCollectionRuleAssociation(context, dcr, vmId);
+            });
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Traced
+  private void createDataCollectionRuleAssociation(
+      FlightContext context, ApiAzureLandingZoneDeployedResource dcr, String vmId) {
+    getMonitorManager(context)
+        .diagnosticSettings()
+        .manager()
+        .serviceClient()
+        .getDataCollectionRuleAssociations()
+        .createWithResponse(
+            vmId,
+            resource.getVmName(),
+            new DataCollectionRuleAssociationProxyOnlyResourceInner()
+                .withDataCollectionRuleId(dcr.getResourceId()),
+            Context.NONE);
+  }
+
+  @Traced
+  private void addMonitorAgentToVm(FlightContext context, String vmId) {
+    var virtualMachine = getComputeManager(context).virtualMachines().getById(vmId);
+
+    var extension =
+        virtualMachine
+            .update()
+            .defineNewExtension("AzureMonitorLinuxAgent")
+            .withPublisher("Microsoft.Azure.Monitor")
+            .withType("AzureMonitorLinuxAgent")
+            .withVersion("1.22")
```

Comment:
Is there a better way to keep the version information? At the very least a constant, or perhaps a configuration entry if there's a straightforward way to do it.